### PR TITLE
feat: add AutoDriveUSDCReceiver contract for USDC intent payments

### LINKED ISSUE #745: Smart contract: AutoDriveUSDCReceiver.sol
## Context
New ERC20 receiver on Ethereum mainnet mirroring `AutoDriveCreditsReceiver.sol` patterns (Ownable2Step, Pausable, ReentrancyGuard). Unlike the native flow (which reads `receipt.from`), the payer must be in the event because ERC20 payments can be relayed.

## Tasks
- [ ] `packages/contracts/src/AutoDriveUSDCReceiver.sol`:
      - Constructor takes the USDC token address + treasury address.
      - `payIntentWithToken(bytes32 intentId, uint256 amount)` — requires prior ERC20 approve,
        does `usdc.safeTransferFrom(msg.sender, address(this), amount)`, emits
        `IntentTokenPaymentReceived(intentId, address(usdc), amount, msg.sender)`.
      - `sweepToTreasury(uint256 amount)` / `sweepAllToTreasury()` — `onlyOwner`.
      - `pause`/`unpause`, `setTreasury`, ReentrancyGuard on the pay path.
- [ ] Use OpenZeppelin `SafeERC20`. Do NOT assume USDC returns a bool.
- [ ] Foundry tests (`forge test`): happy path, missing approval reverts, paused reverts,
      reentrancy guard, sweep auth, event payload correctness, fee-on-transfer/zero-amount edge cases.
- [ ] Deploy scripts for **Sepolia** (Chronos-equivalent test) and **Ethereum mainnet**.
- [ ] Note the Auto EVM London caveat does NOT apply here (deploying on Ethereum, not Auto EVM) — but pin `evm_version`/solc consistently with the repo.

## Acceptance criteria
- `forge build` + `forge test` green.
- Deployed to Sepolia; address + ABI recorded for the frontend/backend issues.
- Permit2/EIP-2612 explicitly deferred (note in code/issue for a future iteration).

### PR DESCRIPTION
**Pull request by EmilFattakhov**
**Part of #742 (USDC Payments for PayWithAI3 epic).**
**Closes #745 — Smart contract: AutoDriveUSDCReceiver.sol.**

Step 3: the ERC20 (USDC) receiver on Ethereum, mirroring `AutoDriveCreditsReceiver` (`Ownable2Step` + `ReentrancyGuard` + `Pausable`). **Independent of #755/#756** — Solidity/Foundry only, so this branches from `main` and can merge on its own.

## Contract — `AutoDriveUSDCReceiver.sol`
- **Immutable USDC token** set in the constructor (redeploy for a different token/chain — no owner repointing).
- `payIntentWithToken(intentId, amount)`: `SafeERC20.safeTransferFrom` + emits `IntentTokenPaymentReceived(intentId, token, amount, payer)`.
  - `payer` travels **in the event** (not read from the tx sender) because ERC20 payments can be relayed.
  - `amount` is the **actual balance delta received**, so a fee-on-transfer token can't make the backend over-credit.
  - Event matches the ABI locked in `@auto-drive/models` in #755 (`intentId` + `payer` indexed).
- **Owner-gated** `sweep(to, amount)` / `sweepAll(to)` to an arbitrary recipient (a swapper later, per Option C). **Callable while paused** so the owner can always rescue funds. (Differs from the native receiver's permissionless-to-fixed-treasury sweep — see design decisions.)
- `pause`/`unpause`; `nonReentrant` on pay + sweep paths.
- **No `minimumBalance`** — that's an Auto-EVM existential-deposit concern, meaningless for an ERC20 balance on Ethereum.

## Design decisions (asked up front)
| Decision | Choice |
| --- | --- |
| Sweep model | Owner-gated flexible `sweep(to, amount)` (B-ready for a swapper) |
| Token address | Immutable, set in constructor |
| Deploy | Foundry script + README |

## Tests — **19, all passing** (`forge test`)
Happy path · missing approval · missing balance · zero amount · paused · fee-on-transfer emits actual received · **reentrancy guard reverts** · sweep onlyOwner + zero-recipient/zero-amount/insufficient-balance · sweepAll happy + empty · sweep-while-paused · pause/unpause auth · two-step ownership · immutable token · zero-token constructor revert. Adds `mocks.sol` (`MockERC20` 6dp, `FeeOnTransferERC20`, `ReentrantERC20`).

## Deploy
`script/DeployAutoDriveUSDCReceiver.s.sol` (env: `PRIVATE_KEY`, `USDC_TOKEN_ADDRESS`, optional `OWNER`) for Sepolia + Ethereum mainnet, documented in the README.

## Verification
- `forge build` clean (Solc 0.8.28; the only warnings are pre-existing in `test/utils.sol`).
- `forge test`: **37/37 passing** (19 new + 18 existing native-contract tests).

## Deferred
Permit2 / EIP-2612 (one-signature approve+pay) intentionally left for a later iteration; v1 is two transactions (approve + pay).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by EmilFattakhov** [comment]
bugbot run

**Review by jim-counter — COMMENTED** [review]
Have we performed any targeted security review on this new contract?

**Comment by EmilFattakhov** [comment]
I've ran two security audits with ChatGPT and Claude, both haven't found any major issues with it. While I don't ever see us using `renounceOwnership` report by GPT 5.5 Pro highlighted that it's inherited and might lead to funds stranding. 

Other than that - no issues found, please find both of the reports below: 

[auto_drive_usdc_receiver_security_audit_pr766.pdf](https://github.com/user-attachments/files/29479861/auto_drive_usdc_receiver_security_audit_pr766.pdf)

[AutoDriveUSDCReceiver_Security_Audit.pdf](https://github.com/user-attachments/files/29479867/AutoDriveUSDCReceiver_Security_Audit.pdf)

File: packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * Required env vars:
+ *   PRIVATE_KEY        - deployer key (hex, 0x-prefixed)
+ *   USDC_TOKEN_ADDRESS - ERC20 (USDC) address on the target chain
+ * Optional:
+ *   OWNER              - initial owner; defaults to the deployer address.
+ *                        Use a multisig for production.
+ *
+ * Known USDC addresses:
+ *   Ethereum mainnet: 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48
+ *   Sepolia:          deploy/point at your test USDC token
```
Comment (jim-counter) [inline packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol]: As per https://developers.circle.com/stablecoins/usdc-contract-addresses this could default to `0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238` if not supplied.

File: packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *   forge script script/DeployAutoDriveUSDCReceiver.s.sol \
+ *     --rpc-url "$ETH_RPC_URL" --broadcast --verify
+ */
+contract DeployAutoDriveUSDCReceiver is Script {
+    function run() external returns (AutoDriveUSDCReceiver receiver) {
+        uint256 deployerKey = vm.envUint("PRIVATE_KEY");
+        address tokenAddress = vm.envAddress("USDC_TOKEN_ADDRESS");
+        address initialOwner = vm.envOr("OWNER", vm.addr(deployerKey));
+
+        require(tokenAddress != address(0), "USDC_TOKEN_ADDRESS is zero");
```
Comment (jim-counter) [inline packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol]: Security audit suggests also checking the address is a contract.

File: packages/contracts/src/AutoDriveUSDCReceiver.sol [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+/**
+ * @title AutoDriveUSDCReceiver
+ * @notice ERC20 (USDC) counterpart of AutoDriveCreditsReceiver, for paying
+ *         credit-purchase intents with a token on Ethereum. The backend watches
+ *         IntentTokenPaymentReceived and grants credits at the intent's locked
+ *         USD price (deferred-conversion / "Option C").
+ * @dev Differs from the native receiver in three deliberate ways:
+ *      - Accepts a single, immutable ERC20 token (redeploy for a different
+ *        token/chain) rather than native value.
+ *      - Emits the `payer` in the event instead of relying on the tx sender,
```
Comment (jim-counter) [inline packages/contracts/src/AutoDriveUSDCReceiver.sol]: This ensures we are recording the account that should receive any refunds rather than intermediary addresses?

File: packages/contracts/src/AutoDriveUSDCReceiver.sol [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    );
+    event Swept(address indexed caller, address indexed to, uint256 amount);
+
+    error InvalidAmount(uint256 amount);
+    error InvalidToken(address token);
+    error InvalidRecipient(address recipient);
+    error InsufficientBalance(uint256 balance, uint256 amount);
+
+    constructor(address initialOwner, IERC20 token_) Ownable(initialOwner) {
+        if (address(token_) == address(0)) {
```
Comment (jim-counter) [inline packages/contracts/src/AutoDriveUSDCReceiver.sol]: Suggest we add the contract check from the audit:

```
if (address(token_) == address(0) || address(token_).code.length == 0) {
  revert InvalidToken(address(token_));
}
```

File: packages/contracts/src/AutoDriveUSDCReceiver.sol [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+}
```
Comment (jim-counter) [inline packages/contracts/src/AutoDriveUSDCReceiver.sol:148]: Suggest we override the renounce feature we inherit. From the audit:
```
error RenounceOwnershipDisabled();

function renounceOwnership() public view override onlyOwner {
  revert RenounceOwnershipDisabled();
}
```

**Review by jim-counter — CHANGES_REQUESTED** [review]
The audit surfaced concerns about us capturing the true payer when payment is made through intermediary contracts. I _think_ we have a test that explicitly refutes that though there could be scenarios I'm not aware of. Warrants a closer look to ensure we know which account should get any refunds due.

File: packages/contracts/src/AutoDriveUSDCReceiver.sol (hunk 3 above)
Comment (EmilFattakhov) [reply packages/contracts/src/AutoDriveUSDCReceiver.sol]: Yes — payer is msg.sender, the account the USDC is actually pulled from, so the event records who funded the payment (the refund target) rather than tx.origin, which could just be whoever initiated a relayed transaction.

File: packages/contracts/src/AutoDriveUSDCReceiver.sol (hunk 4 above)
Comment (EmilFattakhov) [reply packages/contracts/src/AutoDriveUSDCReceiver.sol]: Added 👍

File: packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol (hunk 2 above)
Comment (EmilFattakhov) [reply packages/contracts/script/DeployAutoDriveUSDCReceiver.s.sol]: Added 👍

File: packages/contracts/src/AutoDriveUSDCReceiver.sol (hunk 5 above)
Comment (EmilFattakhov) [reply packages/contracts/src/AutoDriveUSDCReceiver.sol:148]: Good idea, implemented 👍

**Comment by EmilFattakhov** [comment]
Fable 5 security audit: 
[AutoDriveUSDCReceiver_Security_Audit_6593feac.pdf](https://github.com/user-attachments/files/29578437/AutoDriveUSDCReceiver_Security_Audit_6593feac.pdf)

**Review by jim-counter — APPROVED** [review]
LGTM. Thanks for the tweaks!

Nice to see Fable 5 found no issues.
